tests/contrib/operators/test_kubernetes_operator.py

```diff
@@ -46,16 +48,165 @@ class KubernetesJobOperatorTest(unittest.TestCase):
     """
 
     def get_kubernetes_job_operator_task_instance(self):
-        task = KubernetesJobOperator(job_yaml_string=self.YAML,
-                                     sleep_seconds_between_polling=0,
-                                     task_id=self.TASK_ID,
-                                     dag=self.DAG)
+        task = KubernetesJobOperator.from_job_yaml(
+            job_yaml_string=self.YAML,
+            sleep_seconds_between_polling=0,
+            task_id=self.TASK_ID,
+            dag=self.DAG)
 
         task_instance = TaskInstance(task=task,
                                      execution_date=datetime.utcnow() - timedelta(days=1))
 
         return task_instance
 
+    def test_create_from_yaml(self):
+        task = KubernetesJobOperator.from_job_yaml(
+            job_yaml_string=self.YAML,
+            sleep_seconds_between_polling=0,
+            task_id=self.TASK_ID,
+            dag=self.DAG)
+
+        job_name, job_yaml = task.create_job_yaml(dict(execution_date=datetime.utcnow()))
+        self.assertTrue(job_name.startswith('test-job-name-'))
+        parsed = yaml.safe_load(job_yaml)
+        self.assertEqual(1, len(parsed['spec']['template']['spec']['containers']))
+
+        c_zero = parsed['spec']['template']['spec']['containers'][0]
+        self.assertEqual(c_zero['command'][0], 'test-container-command')
+        self.assertNotIn('args', c_zero)
+        self.assertEqual(c_zero['image'], 'test-container-image')
+        self.assertEqual(c_zero['name'], 'test-container-name')
+
+        env_dict = dict([(e['name'], e['value']) for e in c_zero['env']])
+        self.assertEqual('dag-id', env_dict['AIRFLOW_DAG_ID'])
+        self.assertEqual('/test-service-account-secret-name/key.json', env_dict['GOOGLE_APPLICATION_CREDENTIALS'])
+        self.assertEqual('1.10.0dev0+incubating', env_dict['AIRFLOW_VERSION'])
```

Comment:
do we need to test this? (thinking about how someone will have to update assertions if we update the version)